Tests/MongoSwiftTests/ConnectionStringTests.swift

```diff
@@ -346,16 +345,15 @@ final class ConnectionStringTests: MongoSwiftTestCase {
             sleep(5) // sleep to allow heartbeats to occur
         }
 
-        // in case there's an uneven number of events (i.e. client was closed mid-heartbeat) drop any
```

Comment:
in the streaming protocol, heartbeat started events are fired as soon as the previous heartbeat succeeds, so this needed to be updated to check the time between successes instead of between the last successes instead of between the last success and the previous heartbeat's start.